### PR TITLE
Message definitions for on-board companion computer services

### PR DESCRIPTION
**Pull request by magdaFTd**
New messages to support reporting of companion computers statuses & diagnostics, as well as service requests. This is meant to support a ROS inspired communication between GCS and Companion Computer via mavlink including running on-request checks of custom logging, internet/VPN connection (like cellular link) and broadcast like status reporting of the companion computer application status.

### CODE REVIEW

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -998,8 +998,126 @@
       <entry value="10" name="TRACKER_MODE_AUTO"/>
       <entry value="16" name="TRACKER_MODE_INITIALIZING"/>
     </enum>
+    <enum name="COMPANION_BASIC_STATUS_TYPE">
+      <description>Basic status messages types published by the companion computer. Each field is of type uint8_t.</description>
+      <entry value="00" name="COMPANION_STS_EMPTY"/>
```
Comment (hamishwillee) [inline message_definitions/v1.0/ardupilotmega.xml]: Every entry should have a description - just pull in your data comments. This is then used in auto-generated documentation.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 2]
```diff
@@ -998,8 +998,126 @@
...
+      <entry value="51" name="COMPANION_SRV0_RESERVED1"/>
+      <entry value="60" name="COMPANION_SRV0_CELLULAR_LINK_ADD_IP_PORT"/>
+      <entry value="61" name="COMPANION_SRV0_CELLULAR_LINK_LIST_BY_ID"/>
+      <entry value="62" name="COMPANION_SRV0_CELLULAR_LINK_CHECK_BY_ID"/>
+      <entry value="63" name="COMPANION_SRV0_CELLULAR_LINK_REMOVE_BY_ID"/>
+      <entry value="64" name="COMPANION_SRV0_MAX_LIMIT_UINT64"/>
+    </enum>
   </enums>
   <messages>
+    <message id="11800" name="COMPANION_STATUS_REPORT">
```
Comment (hamishwillee) [inline message_definitions/v1.0/ardupilotmega.xml]: Why these IDs?

**Comment by hamishwillee** [comment]
@auturgy Do you know anything about these types? Generally I'd expect these to go into the ArduPilot fork and then be pushed up - that way I'd know that they were expected part of the ArduPilotMega message set.

More generally, the concept of companion messages is something that deserves a lot of thought. I'm not sure that common.xml would be willing to take this set in future without an RFC/lot more discussion.

**Comment by auturgy** [comment]
I haven’t had any visibility of this.
You’re correct: changes to Ardupilotmega should go via the ArduPilot repo first.
Also: https://github.com/mavlink/mavlink/pull/1014 has a related intent.

**Comment by hamishwillee** [comment]
THanks @auturgy . @magdaFTd I'm closing this as changes to dialects should be made first in the dialect project repo, and then pushed here. This ensures that we don't add items to dialects that are not wanted by the dialect.
Very happy to discuss companion messages in common.xml too, but that sort of discussion should probably start as an RFC.

**Pull request closed without merging**